### PR TITLE
libgit2-glib: m1-support add in rpath to build on Apple Silicon

### PR DESCRIPTION
**Pull request by areese**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by carlocab** [comment]
Let's wait for feedback from upstream on your patch.

File: Formula/libgit2-glib.rb [hunk 1]
```diff
@@ -27,6 +27,12 @@ class Libgit2Glib < Formula
...
   depends_on "libgit2"
 
+  patch do
+    # https://gitlab.gnome.org/GNOME/libgit2-glib/-/merge_requests/29
+    url "https://gitlab.gnome.org/GNOME/libgit2-glib/-/commit/e5e61a523508851daa6e54333c2747b7807496fb.diff"
+    sha256 "6b311f239a872b102a018b1ed14a72fdad18ca4e2c47de3c2b6b2e9e9b66a69d"
+  end
+
   def install
     mkdir "build" do
```
Comment (carlocab) [inline Formula/libgit2-glib.rb:37]: I also wonder if we can do without the patch by doing
```ruby
ENV.append "LDFLAGS", "-Wl,-rpath,#{rpath}"
```

Comment (areese) [reply Formula/libgit2-glib.rb:37]: I’ll try that
I know meson was pretty interesting to figure out

**Comment by areese** [comment]
Linking upstream here: https://gitlab.gnome.org/GNOME/libgit2-glib/-/merge_requests/29

File: Formula/libgit2-glib.rb (hunk 1 above)
Comment (areese) [reply Formula/libgit2-glib.rb:37]: ok that worked, lemme push with that instead it seems preferable.

Comment (areese) [reply Formula/libgit2-glib.rb:31]: @carlocab changed to this method

**Review by carlocab — APPROVED** [review]

**Pull request merged**